src/super_gradients/training/utils/checkpoint_utils.py

```diff
@@ -156,6 +161,1229 @@ def read_ckpt_state_dict(ckpt_path: str, device="cpu") -> Mapping[str, torch.Ten
         return state_dict
 
 
+class DefaultCheckpointSolver:
+    """
+    Default checkpoint solver class - has same behavior as PyTorch's default checkpoint solver.
+    """
+
+    def __call__(self, model_state_dict: Mapping[str, Tensor], checkpoint_state_dict: Mapping[str, Tensor]) -> Mapping[str, Tensor]:
+        new_ckpt_dict = {}
+        for (ckpt_key, ckpt_val), (model_key, model_val) in zip(checkpoint_state_dict.items(), model_state_dict.items()):
+
+            if ckpt_val.shape != model_val.shape:
+                raise ValueError(f"ckpt layer {ckpt_key} with shape {ckpt_val.shape} does not match {model_key}" f" with shape {model_val.shape} in the model")
+            new_ckpt_dict[model_key] = ckpt_val
+        return {"net": new_ckpt_dict}
+
+
+class YoloXCheckpointSolver:
+    @classmethod
+    def generate_mapping_table(cls) -> Mapping[str, str]:
+        """
+        Helper method to generate mapping table between olx YoloX checkpoints and the current YoloX layer names.
+        :return: A mapping dictionary {checkpoint_key: model_key}
+        """
+        from super_gradients.common.object_names import Models
+        from super_gradients.training import models
+
+        all_mapping_keys = {}
+        model_names = [Models.YOLOX_N, Models.YOLOX_T, Models.YOLOX_S, Models.YOLOX_M, Models.YOLOX_L]
+
+        for model_name in model_names:
+            model_url = MODEL_URLS[model_name + "_coco"]
+            state_dict = load_state_dict_from_url(model_url, progress=True, map_location="cpu")
+
+            model = models.get(model_name, num_classes=80)
+            model_state_dict = model.state_dict()
+            checkpoint_state_dict = maybe_remove_module_prefix(state_dict["net"])
+            new_sd = {
+                k: v
+                for k, v in checkpoint_state_dict.items()
+                if k not in {"stride", "_head.anchors._anchors", "_head.anchors._anchor_grid", "_head.anchors._stride", "_head._modules_list.14.stride"}
+            }
+
+            for (model_key, model_value), (checkpoint_key, checkpoint_value) in zip(model_state_dict.items(), new_sd.items()):
+                if model_value.size() == checkpoint_value.size() and model_key.split(".")[-1] == checkpoint_key.split(".")[-1]:
+                    if checkpoint_key in all_mapping_keys:
+                        assert all_mapping_keys[checkpoint_key] == model_key
+                    all_mapping_keys[checkpoint_key] = model_key
+                    print(f'"{checkpoint_key}": "{model_key}", ')
```

Comment:
Maybe `log` instead of `print`?

Reply:
Fixed in 78f42e791c337bd10e8643e226864ba271e354ac